### PR TITLE
pep518: purge legacy numpy workaround in setup.py

### PR DESCRIPTION
**Pull request by bjlittle**
### Description

This PR addresses some long outstanding technical debt, which thanks to [pep518](https://peps.python.org/pep-0518/), we can finally tidy-up and simplify.

As a side-effect, developers can now easily build `cf-units` with `python setup.py build_ext --inplace`.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,16 +1,21 @@
...
 from setuptools import Command, Extension, setup
 
 # Default to using cython, but use the .c files if it doesn't exist
 try:
     from Cython.Build import cythonize
 except ImportError:
-    cythonize = False
+    wmsg = "Cython unavailable, unable to cythonize cf-units extensions!"
+    warnings.warn(wmsg)
+    cythonize = None
```
Comment (trexfeathers) [inline setup.py:18]:
```suggestion
    cythonize = False
```

Previous was clearer in IMO

File: setup.py [hunk 2]
```diff
@@ -97,28 +103,11 @@ def get_package_data():
...
-            self.include_dirs.append(numpy.get_include())
-
-    return build_ext(pars)
-
+cython_coverage_enabled = (
+    environ.get("CYTHON_COVERAGE", None) or FLAG_COVERAGE in sys.argv
+)
 
-if FLAG_COVERAGE in sys.argv or environ.get("CYTHON_COVERAGE", None):
+if cythonize and cython_coverage_enabled:
```
Comment (trexfeathers) [inline setup.py:110]: Why has this changed from `or` to `and`?

**Review by trexfeathers — COMMENTED** [review]
I started checking over this but I'm struggling to understand the context so I'll have to stop and leave the comments I had so far.

Something to come back to (or that someone else can take over)

File: setup.py [hunk 3]
```diff
@@ -1,16 +1,21 @@
...
 from distutils.sysconfig import get_config_var
 from os import environ
 from pathlib import Path
 from shutil import copy
 
+# safe to import numpy here thanks to pep518
+import numpy as np
 from setuptools import Command, Extension, setup
 
 # Default to using cython, but use the .c files if it doesn't exist
```
Comment (pelson) [inline setup.py:12]: Why would we want to have a build-time dependency on numpy, but an optional build-time dependency on Cython. Surely we can now move to a world where we unconditionally depend on Cython for building from source (and sdist)?

**Pull request closed without merging**